docs/reference/api/platform/CriticalSectionLock.md

```diff
@@ -1,3 +1,17 @@
 ## CriticalSectionLock
 
+The CriticalSectionLock class provides a mechanism to access a resource without interruption. With the `lock` API, you can enter critical section with interrupts disabled. The `unlock` API is the exit from critical section, and the state of interrupts is restored upon exit. Nesting of critical section is supported, and interrupts are enabled only when we exit from last nested critical section.
```

Comment:
Query: Who or what restores the state of interrupts upon exist? Also, who or what enables interrupts only when we exist from the last critical section?

Reply:
Destructor - restores the state of interrupts/
Destructor